src/main.cpp

```diff
@@ -1396,7 +1402,7 @@ bool CheckInputs(const CTransaction &tx,
                 // failures through additional data in, eg, the coins being
                 // spent being checked as a part of CScriptCheck.
                 const CScript &scriptPubKey = coin.out.scriptPubKey;
-                const CAmount amount = coin.out.nValue;
+                const CAmount &amount = coin.out.nValue;
```

Comment:
similarly here, if we copy the script and amount, we'll be able to release the lock, and let coin fall out of scope.  This means that the lock won't be held during the script execution which will allow us to do it in parallel.  Since script execution generally contains sig validations, its pretty certain that for almost all scripts the copy time is trivial compared to script execution.

CScript scriptPubKey;
CAmount amount;
{
LOCK(inputs.cs_utxo);
￼const Coin &coin = inputs.AccessCoin(prevout);		￼                  
￼if (coin.IsSpent()) LOGA("ASSERTION: no inputs available\n");		￼                      assert(!coin.IsSpent());
scriptPubKey = coin.out.scriptPubKey;
amount = coin.out.nValue;
}

Reply:
yeah, i was trying to avoid the copy but you're right, it's not worth it.  i'll update it...